api/src/routes/register.js

```diff
@@ -0,0 +1,80 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const config = require('config');
+const jwt = require('jsonwebtoken');
+const router = express.Router();
+const secret = config.get('secret');
+const Joi = require('@hapi/joi');
+
+const userSchema = mongoose.Schema({
+    firstName: String,
+    lastName: String,
+    email: String,
+    phone: Number,
+    birthday: Date,
+    created: {
+        type: Date,
+        default: Date.now(),
+    },
+    role: String,
+    password: String,
+    password_repeat: String,
+});
+
+const UserModel = mongoose.model('User', userSchema);
+
+const userValidateSchema = Joi.object({
+    firstName: Joi.string().min(3).max(30).required(),
+    lastName: Joi.string().min(3).max(30).required(),
+    email: Joi.string().email().required(),
+    phone: Joi.string().max(12).required(),
+    created: Joi.date().max('1-1-2050').iso(),
+    role: Joi.string().required(),
+    password: Joi.string().regex(/[a-zA-Z0-9]{6,10}/).required(),
+    password_repeat: Joi.ref('password'),
+});
+
+router.post('/register', (req, res) => {
+    const user = req.body;
+
+    const { value, error } = userValidateSchema.validate(user);
+
+    if (error) {
+        const errors = error.details;
+        res.json(errors);
+        return;
+    }
+
+    const userModel = new UserModel(value);
```

Comment:
This seems a little bit confusing, you have a `UserModel` that you got from the DB and here you create another `userModel`, so please consider to change this naming not to accidentally mix something up.